test/tools/yulInterpreter/EVMInstructionInterpreter.cpp

```diff
@@ -258,6 +271,22 @@ u256 EVMInstructionInterpreter::eval(
 			);
 		logTrace(_instruction, arg);
 		return 0;
+	case Instruction::MCOPY:
+	{
+		// TODO: What if write is valid but read is not? Should we revert msize update?
+		bool writeIsValid = accessMemory(arg[0], arg[2]);
+		bool readIsValid = accessMemory(arg[1], arg[2]);
+		if (writeIsValid && readIsValid)
+			copyZeroExtended(
+				m_state.memory,
+				m_state.memory,
+				static_cast<size_t>(arg[0]),
+				static_cast<size_t>(arg[1]),
+				static_cast<size_t>(arg[2])
+			);
```

Comment:
Does this conform to specification on overlapping source and destination?

Reply:
Right, it was broken and I did not have a test for the overlapping case. Fixed.